File: python/Motorcycle.py
# jenisMotor
# kapasitasTangki

from Vehicle import Vehicle

class Motorcycle(Vehicle):
    __jenisMotor = ""
    __kapasitasTangki = ""

    def __init__(self, jenisMotor = "", kapasitasTangki = ""):
        self.__jenisMotor = jenisMotor
        self.__kapasitasTangki = kapasitasTangki

    # getter
        
    def getjenisMotor(self):
        return self.__jenisMotor
    
    def getkapasitasTangki(self):
        return self.__kapasitasTangki
    
    # setter
    
    def setjenisMotor(self, jenisMotor):
        self.__jenisMotor = jenisMotor

    def setkapasitasTangki(self, kapasitasTangki):
        self.__kapasitasTangki = kapasitasTangki

File: c++/Motorcycle.cpp
#pragma once
#include <iostream>
#include <string>
#include "Vehicle.cpp"
using namespace std;

class Motorcycle : public Vehicle {
private:
    string jenisMotor;
    string kapasitasTangki;

public:
    Motorcycle(string jenisMotor = "", string kapasitasTangki = "") {
        this->jenisMotor = jenisMotor;
        this->kapasitasTangki = kapasitasTangki;
    }

    string getJenisMotor() {
        return jenisMotor;
    }

    string getKapasitasTangki() {
        return kapasitasTangki;
    }

    void setJenisMotor(string jenisMotor) {
        this->jenisMotor = jenisMotor;
    }

    void setKapasitasTangki(string kapasitasTangki) {
        this->kapasitasTangki = kapasitasTangki;
    }
};
